staramr/detection/AMRDetection.py

```diff
@@ -122,6 +131,59 @@ def run_amr_detection(self, files, pid_threshold, plength_threshold_resfinder, p
         self._detailed_summary_dataframe = self._create_detailed_amr_summary(files, self._resfinder_dataframe,
                                                                              self._pointfinder_dataframe,
                                                                              self._plasmidfinder_dataframe)
+                                                                             
+    def _validate_files(self, files: List[str], ignore_invalid_files: bool) -> List[str]:
+        total_files = len(files)
+        removeable_files = []
+
+        for file in files:
+            # Check if the file is not a directory
+            if os.path.isdir(file):
+                if ignore_invalid_files:
+                    logger.warning('--ignore-invalid-files is set, skipping directory {}'.format(file))
+                    removeable_files.append(file)
+                else:
+                    raise Exception('Directory {} is invalid, please use --ignore-invalid-files to skip over this directory'.format(file))
+            else:
+                # Will raise an error if the input returns an empty generator on non-FASTA files, returns a boolean
+                validInput = any(SeqIO.parse(file, "fasta"))
+
+                if not validInput:
+                    if ignore_invalid_files:
+                            logger.warning('--ignore-invalid-files is set, skipping file {}'.format(file))
+                            removeable_files.append(file)
+                    else:
+                        raise Exception('File {} is invalid, please use --ignore-invalid-files to skip over invalid input files'.format(file))
+            
+        # Check to see if the invalid file is not the only file in the directory
+        if total_files == len(removeable_files):
+            raise Exception('Cannot produce output due to no valid input files')
+
+        # Remove the skipped files
+        if ignore_invalid_files:
+            for file in removeable_files:
+                files.remove(file)
+
+        # Check if there are any duplicate sequence id's in the valid files
+        for file in files:
+            record = []
+            # Store all the sequence id's in a list
+            for sequence in SeqIO.parse(file, "fasta"):
+                record.append(sequence.id)
+
+            duplicates = []
+
+            # Each sequence contains a tuple (sequence id, frequency)
+            for sequence in (Counter(record)).items():
+                if sequence[1] > 1:
+                    # We want the sequence id's that are duplicates
+                    duplicates.append(sequence[0])
+
+            # Raise an error if there's a duplicate in the file
+            if len(duplicates) > 0:
+                raise Exception('File {} contains the following duplicate sequence IDs: {}'.format(file, duplicates))
```

Comment:
Looks good. I like how you will print all duplicates instead of just the first one found :+1:

Reply:
Thanks! :smile: